### DISCUSSION TITLE
mysterious out.aar artifacts

### DISCUSSION THREAD

**Discussion by carlleeswanson** [codemagic-ci-cd/codemagic-docs#2352] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

All CodeMagic builds of a new App/Workflow (copied and edited from another similar app/workflow) are producing about 15 "out.aar" files that are listed as downloadable artifacts.

All of our apps are completely Flutter-based, not just using Flutter for a particular module.

Do you know what these "out.aar" files are, where in the Android build process are they being created, and/or why are they being saved?

### Expected results

We expect the CM build to only produce the following artifacts:
- an apk
- an ipa
- some runner zipfiles (which we don't actually need either - so insights on how to suppress those would also be helpful)

### Actual results

In addition to the above files, one workflow produces 15+ different "out.aar" artifacts.

### Build id (optional)

65b804d6ca74d6eba4ec8f7a

**Comment by himesh-cm** [comment]
Hi @carlleeswanson, I cannot reproduce the same. Did you check the contents of the `out.rar` file?

**Reply by carlleeswanson** [reply]
@himesh-cm, I do not see any `out.rar` files. I do see many `out.aar` files, which I am unable to open. They are created by the codemagic build process. You can see them by referencing the BuildID that I included in my original problem description.
- what are these files?
- why are they being saved?

**Reply by himesh-cm** [reply, marked as answer]
Hi @carlleeswanson, 
AAR (Android Archive) binaries are beneficial when integrating Flutter modules into existing Android applications. Although, if you are not using Flutter modules, it is possible that some plugins used in a fully Flutter-based app may unintentionally generate AAR artifacts. This can occur when plugins require compilation into AAR files as part of their functionality, due to the inclusion of native Android code or resources.

As these are binary files, they are automatically exported once the build is successful.
